lib/services/database.go

```diff
@@ -140,14 +141,76 @@ func NewDatabaseFromRDSCluster(cluster *rds.DBCluster) (types.Database, error) {
 	return types.NewDatabaseV3(types.Metadata{
 		Name:        aws.StringValue(cluster.DBClusterIdentifier),
 		Description: fmt.Sprintf("Aurora cluster in %v", metadata.Region),
-		Labels:      labelsFromRDSCluster(cluster, metadata),
+		Labels:      labelsFromRDSCluster(cluster, metadata, RDSEndpointTypePrimary),
 	}, types.DatabaseSpecV3{
 		Protocol: engineToProtocol(aws.StringValue(cluster.Engine)),
 		URI:      fmt.Sprintf("%v:%v", aws.StringValue(cluster.Endpoint), aws.Int64Value(cluster.Port)),
 		AWS:      *metadata,
 	})
 }
 
+// NewDatabaseFromRDSClusterReaderEndpoint creates a database resource from an RDS cluster reader endpoint (Aurora).
+func NewDatabaseFromRDSClusterReaderEndpoint(cluster *rds.DBCluster) (types.Database, error) {
+	metadata, err := MetadataFromRDSCluster(cluster)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return types.NewDatabaseV3(types.Metadata{
+		Name:        fmt.Sprintf("%v-%v", aws.StringValue(cluster.DBClusterIdentifier), string(RDSEndpointTypeReader)),
+		Description: fmt.Sprintf("Aurora cluster in %v (%v endpoint)", metadata.Region, string(RDSEndpointTypeReader)),
+		Labels:      labelsFromRDSCluster(cluster, metadata, RDSEndpointTypeReader),
+	}, types.DatabaseSpecV3{
+		Protocol: engineToProtocol(aws.StringValue(cluster.Engine)),
+		URI:      fmt.Sprintf("%v:%v", aws.StringValue(cluster.ReaderEndpoint), aws.Int64Value(cluster.Port)),
+		AWS:      *metadata,
+
+		// overwrite server name with the primary endpoint name as the rds
+		// instances may not have up-to-date certs that contains reader/custom
+		// endpoints
+		TLS: types.DatabaseTLS{
+			ServerName: aws.StringValue(cluster.Endpoint),
+		},
+	})
+}
+
+// NewDatabasesFromRDSClusterCustomEndpoints creates database resources from RDS cluster custom endpoints (Aurora).
+func NewDatabasesFromRDSClusterCustomEndpoints(cluster *rds.DBCluster) (types.Databases, error) {
+	metadata, err := MetadataFromRDSCluster(cluster)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	var databases types.Databases
+	for _, endpoint := range cluster.CustomEndpoints {
+		endpointName, err := parseRDSCustomEndpoint(aws.StringValue(endpoint))
+		if err != nil {
+			return nil, trace.Wrap(err)
```

Comment:
Instead of bailing on the first error, I would consider collecting all errors and then returning trace.NewAggregate so we can import at least some custom endpoints.